hoomd/hpmc/compute.py

```diff
@@ -7,123 +7,111 @@
 from __future__ import print_function
 
 from hoomd import _hoomd
+from hoomd.operation import Compute
 from hoomd.hpmc import _hpmc
 from hoomd.hpmc import integrate
-from hoomd.operation import Compute
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.data.typeconverter import OnlyTypes
+from hoomd.logging import log
 import hoomd
 
-class free_volume(Compute):
-    R""" Compute the free volume available to a test particle by stochastic integration.
+class FreeVolume(Compute):
+    """Compute the free volume available to a test particle by stochastic integration.
 
     Args:
-        mc (:py:mod:`hoomd.hpmc.integrate`): MC integrator.
-        seed (int): Random seed for MC integration.
+        mc (:py:mod:`hoomd.hpmc.integrate.HPMCIntegrator`): MC integrator.
         type (str): Type of particle to use for integration
         nsample (int): Number of samples to use in MC integration
-        suffix (str): Suffix to use for log quantity
 
-    :py:class`free_volume` computes the free volume of a particle assembly using stochastic integration with a test particle type.
+    :py:class`FreeVolume` computes the free volume of a particle assembly using stochastic integration with a test particle type.
     It works together with an HPMC integrator, which defines the particle types used in the simulation.
     As parameters it requires the number of MC integration samples (*nsample*), and the type of particle (*test_type*)
     to use for the integration.
 
-    Once initialized, the compute provides a log quantity
-    called **hpmc_free_volume**, that can be logged via ``hoomd.analyze.log``.
-    If a suffix is specified, the log quantities name will be
-    **hpmc_free_volume_suffix**.
+    A :py:class`FreeVolume` object can be added to a logger for logging during a simulation,
+    see :py:class:`hoomd.logging.Logger` for more details.
```

Comment:
```suggestion
```
All HOOMD operations can be added to loggers. We don't need to mention that explicitly here.